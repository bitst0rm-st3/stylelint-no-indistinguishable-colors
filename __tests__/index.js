const stylelint = require("stylelint");
const plugin = require("../lib");
const messages = plugin.messages;
const ruleName = plugin.ruleName;
const rule = plugin.rule;

function getOptions(code, rules) {
  return {
    code,
    configBasedir: __dirname,
    config: {
      plugins: "../lib",
      rules
    }
  };
}

describe("stylelint-no-indistinguishable-colors", () => {
  describe("true option", () => {
    it("should enable linting", done => {
      const rules = {
        [ruleName]: true
      };
      const code = `div { color: #000; color: #010101; }`;

      return stylelint
        .lint(getOptions(code, rules))
        .then(result => {
          expect(result.errored).toBe(true);
          expect(JSON.parse(result.output)).toMatchSnapshot();
          return done();
        })
        .catch(error => done(error));
    });
  });
});
