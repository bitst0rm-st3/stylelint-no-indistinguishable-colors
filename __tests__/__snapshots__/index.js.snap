// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stylelint-no-indistinguishable-colors true option should enable linting 1`] = `
Array [
  Object {
    "deprecations": Array [],
    "errored": true,
    "invalidOptionWarnings": Array [],
    "parseErrors": Array [],
    "source": "<input css 1>",
    "warnings": Array [
      Object {
        "column": 20,
        "line": 1,
        "rule": "plugin/stylelint-no-indistinguishable-colors",
        "severity": "error",
        "text": "Unexpected indistinguishable colors \\"#010101\\" and \\"#000\\". (plugin/stylelint-no-indistinguishable-colors)",
      },
    ],
  },
]
`;
